scripts/ci/test-operator

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+# This script transforms an operator dir structured in the style of
+# community-operators into one expected by operator-registry, inserts the
+# scorecard proxy container, and proxy kubeconfig secret, volume, and mount
+# into a CSV, creates CR's from CSV metadata, deploys the operator with the
+# OLM in a local cluster, and runs the SDK scorecard against the operator.
+
+set -ex
+
+for f in "$(cd "$(dirname ${BASH_SOURCE[0]})/.." && pwd)"/lib/*; do
+  . "$f"
+done
+
+# Relative path to the dir containing operator manifests. Usually something
+# like "community-operators/automationbroker".
+[[ -z "$OP_PATH" ]] && OP_PATH="$1"
+if [[ ! -d "$OP_PATH" ]]; then
+  echo "Relative path to operator package '$OP_PATH' does not exist."
+  exit 1
+fi
+# This script assumes versions are written as "v0.0.1" in file names and
+# fields, but OP_VER is expected to not include a "v" prefix, ex. "0.0.1".
+[[ -z "$OP_VER" ]] && OP_VER="$2"
+if [[ -z "$OP_VER" || ! "$OP_VER" =~ ^[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$ ]]; then
+  echo "Operator version '$OP_VER' not valid."
+  exit 1
+fi
+DISTRO_TYPE_UPSTREAM="upstream"
+DISTRO_TYPE_OPENSHIFT="openshift"
+DISTRO_TYPE="${3:-$DISTRO_TYPE_UPSTREAM}"
+if [[ "$DISTRO_TYPE" != "$DISTRO_TYPE_UPSTREAM" && "$DISTRO_TYPE" != "$DISTRO_TYPE_OPENSHIFT" ]]; then
+  echo "DISTRO_TYPE \"$DISTRO_TYPE\" is not valid. Must be one of: \"$DISTRO_TYPE_UPSTREAM\", \"$DISTRO_TYPE_OPENSHIFT\"."
+  exit 1
+fi
+if [[ "$DISTRO_TYPE" == "$DISTRO_TYPE_OPENSHIFT" ]]; then
+  if [[ -z "$KUBECONFIG" || ! -e "$KUBECONFIG" ]]; then
+    echo "KUBECONFIG not available."
+    exit 1
+  fi
+fi
+
+TMP="$(mktemp -d --suffix "_test_pr")"
+trap_add_exit "rm -rf $TMP"
+set -u
+DEPLOY_DIR="${TMP}/deploy"
+PKG_FILE="$(find "$OP_PATH" -name "*.package.yaml" -print -quit)"
+PKG_NAME="$(yq r "$PKG_FILE" "packageName")"
+ABS_BUNDLE_PATH="${DEPLOY_DIR}/${PKG_NAME}/${OP_VER}"
+CR_DIR="${DEPLOY_DIR}/crs"
+mkdir -p "$ABS_BUNDLE_PATH"
+mkdir -p "$CR_DIR"
+
+# Organize expected dir structure for the registry image build.
+cp -a "$(dirname "$OP_PATH")/$(basename "$OP_PATH")"/* "$ABS_BUNDLE_PATH"
+pushd "$DEPLOY_DIR"
+mv "${ABS_BUNDLE_PATH}/${PKG_NAME}.package.yaml" "${PKG_NAME}/"
+NAMESPACE="local"
+CATALOGSOURCE_FILE="${PKG_NAME}.catalogsource.yaml"
+SUBSCRIPTION_FILE="${PKG_NAME}.subscription.yaml"
+SC_PROXY_IMAGE="quay.io/operator-framework/scorecard-proxy:master"
+# Make sure this registry is public before proceeding.
+OP_REGISTRY_IMAGE="quay.io/operatorframework/${PKG_NAME}:${OP_VER}-registry"
+SECRET_FILE="${DEPLOY_DIR}/scorecard.secret.yaml"
+CSV_FILE="$(find "$ABS_BUNDLE_PATH" -name "*.v${OP_VER}.clusterserviceversion.yaml" -print -quit)"
```

Comment:
Some CSVs don't follow this pattern. It may be more general to use `"*${OP_VER}.clusterserviceversion.yaml"` to match a particular CSV file.